syntax = "proto2";

package objstripe;

enum SerializeStrategy {
  kRoot = 0;
  kRootUnrolled = 1;
}

message ProductInfo {
  optional string productName = 1;
  optional string productType = 2;
  optional uint32 flushSize = 3;
  optional uint32 flushMinBytes = 4;
}

message ObjectStripeIndex {
  optional uint32 eventStripeSize = 1;
  optional uint64 totalEvents = 2;
  repeated ProductInfo products = 3;
  repeated bytes packedEventStripes = 4;
  optional SerializeStrategy serializeStrategy = 5;
}

message EventStripe {
  message Event {
    optional uint64 offset = 1;
    optional uint32 run = 2;
    optional uint32 lumi = 3;
    optional uint64 event = 4;
  }

  repeated Event events = 1;
  // TODO: store product flushSize here?
}

message ProductStripe {
  repeated uint32 offsets = 1 [packed = true];
  optional bytes content = 2;
  optional uint64 globalOffset = 3;
}
